src/main/java/seedu/duke/parsers/AddParser.java

```diff
@@ -14,46 +14,55 @@
 
     // Unescaped regex for testing (split into two lines):
-    // \s*(\/t\s+(?<taskName>.+?(?=(\s+-d\s+)|$))(\s+(-d\s+(?<taskDescription>.+)))?|
-    // \/m\s+(?<moduleCode>\w+?(?=(\s+-d\s+)|$))(\s+(-d\s+(?<moduleDescription>.+)))?)
+    // ^\s*(\/t\s+(?<taskName>.+?(?=\s+-d\s+|\s+-t\s+|$))(\s+(-d\s+\"(?<taskDescription>([^\"]*))\")(?=(\s+-t\s+)|$))?
+    // (\s+(-t\s+\"(?<estimatedWorkingTime>([^\"]*))\")(?=(\s+-d\s+)|$))?|\/m\s+(?<moduleCode>\w+?(?=(\s+-d\s+)|$))
+    // (\s+(-d\s+\"(?<moduleDescription>.+)\"))?)
     // TODO: Add support for -mod argument when integrating Task and Module classes with one another
-    private static final String ADD_FORMAT = "\\s*(\\/t\\s+(?<taskName>.+?(?=(\\s+-d\\s+)|$))"
-            + "(\\s+(-d\\s+(?<taskDescription>.+)))?|\\/m\\s+(?<moduleCode>\\w+?(?=(\\s+-d\\s+)|$))"
-            + "(\\s+(-d\\s+(?<moduleDescription>.+)))?)";
+    private static final String ADD_FORMAT = "\\s*(\\/t\\s+(?<taskName>.+?(?=\\s+-d\\s+|\\s+-t\\s+|$))"
+            + "(\\s+(-d\\s+\\\"(?<taskDescription>([^\\\"]*))\\\")(?=(\\s+-t\\s+)|$))?(\\s+(-t\\s+\\\""
+            + "(?<estimatedWorkingTime>([^\\\"]*))\\\"))?|\\/m\\s+"
+            + "(?<moduleCode>\\w+?(?=(\\s+-d\\s+)|$))(\\s+(-d\\s+\\\"(?<moduleDescription>.+)\\\"))?)";
 
     public AddParser() {
         super();
         // See also https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html
         commandFormat = ADD_FORMAT;
         groupNames.add(TASK_NAME);
         groupNames.add(TASK_DESCRIPTION);
+        groupNames.add(TASK_WORKING_TIME);
         groupNames.add(MODULE_CODE);
         groupNames.add(MODULE_DESCRIPTION);
     }
-
     
     @Override
     public Command parseCommand(String userInput) throws ModHappyException {
         HashMap<String, String> parsedArguments = parseString(userInput);
         final String taskName = parsedArguments.get(TASK_NAME);
         final String taskDescription = parsedArguments.get(TASK_DESCRIPTION);
+        final String estimatedWorkingTime = parsedArguments.get(TASK_WORKING_TIME);
         final String moduleCode = parsedArguments.get(MODULE_CODE);
         final String moduleDescription = parsedArguments.get(MODULE_DESCRIPTION);
         if (!taskName.equals(EMPTY_STRING)) {
-            if (!taskDescription.equals(EMPTY_STRING)) {
-                return new AddCommand(taskName, taskDescription, true);
+            if (!estimatedWorkingTime.equals(EMPTY_STRING) && !taskDescription.equals(EMPTY_STRING)) {
+                return new AddCommand(taskName, taskDescription, true, estimatedWorkingTime);
+            } else if (!taskDescription.equals(EMPTY_STRING)) {
+                return new AddCommand(taskName, taskDescription, true, null);
+            } else if (!estimatedWorkingTime.equals(EMPTY_STRING)) {
+                return new AddCommand(taskName, null, true, estimatedWorkingTime);
+            } else {
+                return new AddCommand(taskName, null, true, null);
             }
```

Comment:
Suggest to use a switch to make it looks easier. Also avoid use magic variables such as null, true here. for the taskDescription, I suggest to pass the original taskDescription directly. For the isTask boolean, I suggest to make a method to judge.

Reply:
null and true are actually based on my implementation of the AddCommand constructor if the user did not supply some of the optional arguments. We can discuss this later if you have a better way to implement it.

Reply:
Ok, for the null value of taskDescription, a possible way is to change the default character for missing arguments as null in Parser.java. Then no need to do the EMPTY_STRING check here

Reply:
isEmpty() can also be used to check for null, might make it more readable than just putting the variable name

Reply:
Ive changed all checking for empty string to null fields, including in DeleteParser.java